Task 1 to release

AlexV0rob opened the pull request:
Apply changes from task_1 branch to main. Added functionality:
1) Bot replies to /start and /help commands with certain messages;
2) Bot returns every other message with "You have sent: " prefix.

Commit:
Added MainLogic.java, MainLogicTest.java, pom.xml

Added three new files:
1) MainLogic.java - class that does main bot logic
2) MainLogicTest.java - unit-tests for MainLogic class
3) pom.xml - standard Maven file but with dependencies for JUnit

AlexV0rob committed:
Merge pull request #1 from AlexV0rob/task_1_Vorobyov

Added MainLogic.java, MainLogicTest.java, pom.xml

Commit:
Добавление TelegramBot.java и Main.java

KashKetchup committed:
Merge branch 'task_1' into task_1_Nikita

KashKetchup committed:
Merge pull request #2 from AlexV0rob/task_1_Nikita

Добавление файлов из task_1_Nikita в task_1

AlexV0rob committed:
Update pom.xml

New version of pom.xml:
1) Changed GroupId to ru.urfu
2) Added compiler version
3) Added Apache Commons libraries and JUnit Platform libraries

AlexV0rob committed:
Merge pull request #3 from AlexV0rob/task_1_pom_fix

Update pom.xml

KashKetchup committed:
Update README.md

Обновление описания проекта

KashKetchup committed:
Update README.md

KashKetchup committed:
Update README.md

переписывание Readme

KashKetchup committed:
Update README.md

KashKetchup committed:
Merge pull request #5 from AlexV0rob/KashKetchup-patch-for-README.md

Kash ketchup patch for readme.md

Commit:
Комментарии, Javadoc и реплики бота переведены

Все комментарии, Javadoc и реплики бота в классах MainLogic и
MainLogicTest переведены с английского на русский.

vpyzhyanov commented on src/main/java/org/example/Main.java:
```diff
@@ -0,0 +1,24 @@
+import org.telegram.telegrambots.longpolling.TelegramBotsLongPollingApplication;
+
+/**
+ * 
```
нет описания

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -0,0 +1,84 @@
+package org.example;
+
+/**
+ * Does the main management
```
непонятно что это за управление такое. Конкретизируйте. Слово main лучше не использовать, оно не несёт никакого смысла в этом контексте

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -0,0 +1,84 @@
+ */
+public class MainLogic {
+	/**
+	 * Does the prime management calling auxiliary methods
```
1. тоже общие слова, без описания назначения метода
2. Использована вопросительная форма предложения. Напоминаю, что метод - это глагол (действие), тем более он возвращает не boolean

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -0,0 +1,84 @@
+public class MainLogic {
+	/**
+	 * Does the prime management calling auxiliary methods
+	 * @param input keeps user's input
```
не надо никаких вводных слов типа "переменная содержит". Она для этого и существует, поэтому не надо писать очевидных вещей. А если убрать вводные слова, то останется только имя, которое о себе может само заявить. Например `userInput`

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -0,0 +1,84 @@
+	 */
+	public String processInput(String input) {
+		switch (input) {
+		//If input is a command /start
```
любителе чата GTP будем исключать из группы extra

Настолько очевидные вещи пишут либо нейросети, либо студенты, которые написали код не сами

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -0,0 +1,84 @@
+		String startMessage = "Hello, fellow citizen! I'm a bot about chess. My current options are:\n";
+		
+		//List of abilities
+		startMessage += " - return your messages;\n";
```
1. Надо убрать бесконечное переписывание одной перменной. Тем более смысла нет, можно сразу сформировать нужную строку
2. Используйте многострочный комментарий

vpyzhyanov replied:
> Используйте многострочный комментарий

Не исправлено

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -0,0 +1,84 @@
+		startMessage += " - return your messages;\n";
+		startMessage += " - show help window;\n";
+		
+		//New line for more comfortable reading
```
чат GPT

vpyzhyanov commented on src/main/java/org/example/TelegramBot.java:
```diff
@@ -0,0 +1,58 @@
+    }
+	
+	/**
+	 * Получает на вход update и передаёт управление MainLogic.java
```
Обратите внимание на `@Override` и то где определён метод. Нельзя переопределять описание метода интерфейса, вы разрушаете полиморфизм

vpyzhyanov commented on src/test/java/org/example/MainLogicTest.java:
```diff
@@ -0,0 +1,33 @@
+	 */
+    @Test
+    void startCommandTest() {
+        assertEquals("Hello, fellow citizen! I'm a bot about chess. My current options are:\n - return your messages;\n - show help window;\n\nThis is all by now but the list of abilities will be updated during development. Type /help for more information.", new MainLogic().processInput("/start"));
```
не надо увлекаться однострочным программированием

vpyzhyanov commented on src/test/java/org/example/MainLogicTest.java:
```diff
@@ -0,0 +1,33 @@
+	 */
+    @Test
+    void helpCommandTest() {
+        assertEquals("I can now:\n - return your messages;\n - show help window;\n\nCurrently available commands:\n/help - shows this message\n/start - restarts bot\n\nMore oprions will be available soon.", new MainLogic().processInput("/help"));
```
не надо увлекаться однострочным программированием

vpyzhyanov reviewed (changes requested):
В следующий раз решение сделанное через чат GPT даже смотреть не буду. А вас исключу из группы extra

Commit:
Исправлены замечания:

1)В Main.java изменён javadoc.
2)В TelegramBot.java убрал лишний javadocds

KashKetchup committed:
Update Main.java

Небольшое исправление import org.example

KashKetchup committed:
Merge pull request #9 from AlexV0rob/task_1_Nikita_Fix

Исправлены замечания:

Commit:
Исправления замечаний и перевод Javadoc и комментариев

Код почищен от лишних комментариев.
Оставленные комментарии и Javadoc переведены на русский.
В классе MainLogicTest удалены методы, проверяющие работу
пользовательского ввода /start и /help за ненадобностью.

AlexV0rob committed:
Merge pull request #10 from AlexV0rob/task_2_translation_to_russian

Исправление замечаний по task_1

vpyzhyanov commented on src/main/java/org/example/Main.java:
```diff
@@ -3,10 +3,12 @@
 
 /**
- * 
+ * Пытается создать экземпляр бота 
```
"Пытается" и не получается? Не очень удачная формулировка, какая-то неуверенная. Например так можно написать:  "Основной класс для запуска бота"

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -1,84 +1,69 @@
-		
-		//New line for more comfortable reading
-		startMessage += "\n";
+		String startMessage = "Здравствуй, путник! Я бот о шахматах. Сейчас я умею:\n"
```
1. Переменная не нужна, можно сразу вернуть нужное значение
2. Метод не нужен, он ничего не делает
3. Текст ответа на команду нужно определить в константе, а в методе processInput обратиться сразу к константе.

vpyzhyanov commented on src/main/java/org/example/MainLogic.java:
```diff
@@ -1,84 +1,69 @@
-		
-		//New line for more comfortable reading
-		helpMessage += "\n";
+		String helpMessage = "Сейчас я могу:\n"
```
то же самое

vpyzhyanov reviewed (approved):
Задача принята. Как исправите последние замечания, можете мёрджить

KashKetchup committed:
Update Main.java

Изменения описания класса Main

KashKetchup committed:
Merge pull request #11 from AlexV0rob/task_1_Nikita_Fix

Update Main.java

Commit:
Финальное исправление task_1

Тексты сообщений команд /start и /help вынесены в константы и
преобразованы из строк в текстовые блоки;
Удалены методы startCommand и helpCommand за ненадобностью.

AlexV0rob committed:
Merge pull request #12 from AlexV0rob/task_1_MainLogic_hotfix

Финальное исправление task_1
